app/controllers/sequences_controller.rb

```diff
@@ -114,6 +114,29 @@ def show
 
     @title = "Tryptic peptide analysis of #{@original_sequence}"
 
+    @pept_with_go = 0
+    @pept_with_ec = 0
+    go_counts = Hash.new 0
+    @go_summary = {}
+    @ec_summary = Hash.new 0
+    @entries.each do |entery|
+      go_terms = entery.go_cross_references.map(&:go_term)
```

Comment:
you can request the go_terms directly from the entries

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13729) by @bmesuere on Mon Feb 19 2018 at 14:36._